### ISSUE TITLE
Core/Movement: waypoint system issues [45$]

### ISSUE THREAD

**Issue by minkokomaung** [TrinityCore/TrinityCore#20310]
**Description:**
When you click Inconspicuous Mine Car, you'll be inside Mine Car and Scarlet Miner should pull that mine car toward Light's Point.
Currently it doesn't happen.

http://www.wowhead.com/quest=12701/massacre-at-lights-point

**Branch(es):** 3.3.5

**TC rev. hash/commit:** 62f7debffdb0

**TDB version:**  TDB 3.3.5.63

**Operating system:** Windows 10 x64

**Comment by dikkedeur** [comment]
i can confirm, the mine cart seems not to be working

**Comment by Foereaper** [comment]
Seems to be happening after the following commit:

https://github.com/TrinityCore/TrinityCore/commit/7fff83d67526efff63867d41b9e036a19a9287b3

**Comment by Foereaper** [comment]
Copied from issue 20341:

Several features seem to have stopped working as intended after the above mentioned commit and subsequent updates. I will list things I discover below as I go along:

**Quests:**

- Massacre at Light's Point(12701), the minecart no longer gets pulled down to the boats.
- Territorial Trespass (13051), the dragon no longer lands upon quest progression.

**Instances:**

- Azjol'Nerub, Hadronox keeps resetting as soon as one of the debuffs runs out. Impossible to complete without GM intervention.
- The Culling of Stratholme, Arthas keeps getting stuck on adds and no longer continues to follow his waypoints, impossible to complete without GM intervention.

I will keep this updated with more examples as I find them, but these are the ones I have noticed so far.

**Comment by Infinito47** [comment]
I think a priority tag would suit well; there is a lot broken after 7fff83d

**Comment by ConradBunton** [comment]
Violet hold works fine, for me.

**Comment by Infinito47** [comment]
@ConradBunton On what commit are you?

**Comment by tkrokli** [comment]
Yes, please always include TC commit rev. hash when confirming or disputing issues. A statement about what works or not on TC is almost worthless without the actual local TC rev. hash for comparison.

**Comment by ConradBunton** [comment]
I'm on 20bc7d7

**Comment by Foereaper** [comment]
After debugging Massacre at Light's Point(12701) (see chapter1.cpp, npc_scarlet_miner) I can see that, while Start() is called (EscortAI), the waypoints are never actually started, and WaypointReached() never reached at all, preventing any of the subsequent events from firing.

I have compared this to other scripts that use the same exact way of doing waypoints, and while they work well there, they do not in the above case.

Something else I noticed is that IsSummonedBy() sets carGUID, however after IsSummonedBy() is called, Reset() gets called which clears carGUID. Whether this actually matters with the waypoint part I cannot tell, as it is not used anywhere except further down the line into WaypointReached(), but I cannot see this being the intended functionality.

**Comment by Foereaper** [comment]
Should these issues be split into separate issues to get someone to take a look at them or are we awaiting @ccrs ? Considering how much this impacts different ingame features, any endgame content is close to not viable at this rate.

**Comment by tkrokli** [comment]
Usually, separate issues need separate tickets. But if the underlying cause is the same for all of the issues, there is no need to separate them as long as the solution will be the same for all of them and fix all of them at once.

Post separate tickets for those issues where you believe the solution is different from the other issues, if there are any.

**Comment by Foereaper** [comment]
Alright, sounds good. I guess I'll keep debugging the other issues to provide as much information as possible, to try and determine whether they all fail the same way or if some of them differ from the rest.

**Comment by Dalaspring** [comment]
Hello Trinitycore. We are also experiencing many issues on our server due to this commit. I have created an account here to subscribe to this issue.

Is there any reason why @ccrs has been absent? With the amount of issues right now it's almost impossible to call TrinityCore a reliable source for our server right now.

**Comment by tkrokli** [comment]
@Dalaspring : TrinityCore is an idealistic team effort, both for the sake of learning and utilizing coding knowledge in a familiar environment, especially when it comes to testing the results. Whether or not it creates problems for people using the source as a playable server is not our main concern. The goal here is to learn from earlier mistakes and work towards finding new solutions as well as keeping the result as close to the original Blizzard environment as much as possible. There are limitations, both in how much time the users and TC members are able to set aside for this kind of work, as well as following up new issues and solutions posted by other users.

The bottom line is: if it does not work the way you want it to, you can choose between going back to earlier commits before the current issue appeared, or just use the source as it is.

Most of the TC members experience real life events which may (more often than not) limit the time they can set aside to work on the current source and its issues. All we can do is wait and see how it goes. Whether or not **ccrs** is able to follow up the current issue is something we have to deal with based on eventual feedback from him (or the lack thereof).

**Comment by ccrs** [comment]
thank for your words @tkrokli much appreciated.

> Is there any reason why @ccrs has been absent?

Do you think there is any form of obligation or anything that compels me to followup this minor issues? I and all the collaborators decide how much of our spare time is invested into the repo. Which, as its obvious, now its close to zero in my case.

> With the amount of issues right now it's almost impossible to call TrinityCore a reliable source for our server right now.

ROFL, wont bother to comment that

@Foereaper thank you for the research work, those are all bad scripts that need fine tunning.
The more you find the better 👍 
Though some of the things you are pointing already were messed up before the refactor/rework

**Comment by Foereaper** [comment]
I'll keep updating with what I find as I go along, I only started getting reports on most of them after we updated past the said commit so figured they were all related hehe.

**Comment by Dalgurakk** [comment]
Please allow me a simple comment.
I have always used trinitycore and I have recommended it to many friends always with the prestige of so many years. Until now 3.3.5 is the most successful expansion of wow, in this repository most of the issues are related to this expansion, however the trinitycore members are always focused on the last expansion, right now only a few continue to support this branch. The amazing work of  @Keader , @Treeston , @Killyana , @tkrokli , and @sirikfoll  is not enough. Really there are many issues in 3.3.5. We miss @Shauren , @ariel-  and others. I just ask to TrinityCore that they do not forget us because we feel us TrinityCore too.

**Comment by Treeston** [comment]
wat

**Comment by Jildor** [comment]
please stop putting comments that do not help solve the issue.
Use the forum for these things.

**Comment by Foereaper** [comment]
@ccrs Let me know if there is anything you need a hand with, and if I should continue debugging the above issues for information.

**Comment by Noryad** [comment]
For desperate players, try to get off the horse when the horse comes back and reassemble and go ahead a few fts and repeat the operation, it is not the solution but it can help.

**Comment by Infinito47** [comment]
Any progress on this?

**Comment by tkrokli** [comment]
Please show more patience, @Infinito47 . This issue has not even been open for 3 weeks yet. Unless you have something constructive to contribute with, please don't ask or comment. Remember that those users who may be able to do something about this issue are real people with real lives and that they contribute in their unpaid spare time. Feel free to chat about the issue in the TC forum pages if you like, but not on this page. Unless you see someone posting something useful here or in a linked page, there is no new development or progress on this issue. If you by chance know any good coders who might be able to contribute, feel free to suggest that they sign up for a github account (if they don't have one already) to post something useful here.

Also be aware that unhelpful comments (even my comments) can be removed from these issue tracker pages if needed to keep it in order. (This is **not** a forum.)
- Use the https://community.trinitycore.org/ (TC Forum) pages for non-contributing discussions.
- Visit the [#trinity IRC help channel](https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130200/IRC) if you want simple answers to simple questions.

**Comment by amdwow** [comment]
Oh my god.. what's so hard to assign some other developer to this! Stop releasing updates and focus on this please.. and @ccrs  minor issue? Are you mentally retarded??

**Comment by ariel-** [comment]
> Oh my god.. what's so hard to assign some other developer to this! Stop releasing updates and focus on this please.. and @ccrs minor issue? Are you mentally retarded??

if you're that smart go fix it yourself and make a PR, otherwise just chill the fuck out, developers have a life too.

**Comment by amdwow** [comment]
i dont care if developers have a life. u break something.. u fix it.  Especially with this much impact.. 

I know most of you will not apriciate my answer but come on.. it did not get priority tag on trinity and also there is not single development!!!! Not a  single update or attempt to fix it! wtf!

**Comment by STARRHELD** [comment]
You fix it @amdwow - also you are particularly rude. You should show some more respect since this project is completely free.

**Comment by Keader** [comment]
It's a open source project Dude. Not only "tc team" can fix it.

**Comment by ariel-** [comment]
> i dont care if developers have a life. u break something.. u fix it. Especially with this much impact..

like if were a paid job? what if 'fixing' it breaks another thing? if it were that simple it would be fixed by now

> I know most of you will not apriciate my answer but come on.. it did not get priority tag on trinity and also there is not single development!!!! Not a single update or attempt to fix it! wtf!

you know, there are older issues than this one.

**Comment by Aokromes** [comment]
Test after 7dc97c0

**Comment by ccrs** [comment]
not related

**Comment by Faq** [comment]
Please retest after https://github.com/TrinityCore/TrinityCore/commit/e10d7dd45cfc073c791cd5e5f3f964d14b37df3d

**Comment by Killyana** [comment]
Only some issues are fixed, other ones still happen.

**Comment by ccrs** [comment]
and again, just pointing out, most of the issues were there before the rework commit

**Comment by jonwgary** [comment]
FWIW, this didn't fix the issue with the mine car in Massacre At Light's Point. I didn't test this before the 7fff83d commit, so I have no idea what caused the bug, but it remains.

Thanks to you folks working on this. If I can find some time this weekend, I might try to debug it myself. Even if I don't figure it out, I'm still interested in messing around a bit in this codebase to see if I can help in any way.

**Comment by ogeraisi** [comment]
DK starting quest "Death Comes From On High" is still broken

**Comment by tkrokli** [comment]
@ogeraisi : remember to add your core rev. hash in the message if you post a "confirm" message in #18362 (still open as of the time of my comment).

**Comment by ogeraisi** [comment]
@tkrokli I was told before that the bug is due to the core movement commit which is why I'm posting here and not on a previous or a new bug report but I will confirm it there as well :)

**Comment by Treeston** [comment]
Certain escort/follower quests (iirc minecart is one of them) are broken due to an unrelated AI bug, @Wyreth is working on it.

**Comment by Furrazzlhf** [comment]
Do someone know how could i fix the very long delay when  i wich start the 5 phases in culling of stratholme because arthas start moving 3-4 minutes after we talk him ... (3.3.5 branch)
Thanks a lot for the guy who fix my problem ^^

**Comment by tkrokli** [comment]
@Furrazzlhf : This is not a forum. Use the TC Forum or #trinity @ irc.rizon.net for stuff like that.

**Comment by Foereaper** [comment]
@Furrazzlhf This is part of the waypoint issue, it seems the event supposed to trigger on reached end of waypoint isn't triggering like it should, see Culling of Stratholme and Territorial Trespass (13051)

**Comment by noblist** [comment]
1 ingame fix, non admin, that works, is as soon as the horse turns and starts walking, jump off of the horse then get back on. Doing this enough times does get the quest done, but not a real fix. A temporary until someone who knows programming finds the fixed script error. 

I would post my TC but am responding on mobile at the moment, sorry

**Comment by tje3d** [comment]
Smart scripts with **event_type 40** _(SMART_EVENT_WAYPOINT_REACHED)_ and **action type 15**  _(SMART_ACTION_CALL_AREAEXPLOREDOREVENTHAPPENS)_ are not works.
Is it related to this issue?

**Comment by Foereaper** [comment]
https://github.com/TrinityCore/TrinityCore/issues/20771

See the last comment on this issue, it's definitely related to the rewrite, as the quest in question worked fine before it and does after the mentioned edit, however I'm not sure whether it is related to your question directly.

**Comment by jerbookins** [comment]
1f9f59b4b3b9

In reference to #20882 ([Disclosure](www.wowhead.com/quest=12710/disclosure)), I realized a couple things wrong with the dialogue going through the quest again that could help explain why the quest isn't working.

This is the dialogue as it is currently:

> `1` Malmortis says: Ahh... there you are. The master told us you'd be arriving soon.
`2` Malmortis says: Please, follow me, **$n**. There is much for you to see....
`3` Malmortis says: You should feel honored. You are the first of the master's prospects to be shown our operation.
`4` Malmortis says: The things I show you now must never be spoken of outside Voltarus. The world shall come to know our secret soon enough!
`5` Malmortis says: Here lie our stores of blight crystal, without which our project would be impossible.
`6` Malmortis says: I understand that you are to thank for the bulk of our supply.
`7` Malmortis says: These trolls are among those you exposed on the battlefield. Masterfully done, indeed....
`8` Malmortis says: We feel it best to position them here, where they might come to terms with their impending fate.
`9` Malmortis says: This is their destiny....
`10` Malmortis says: The blight slowly seeps into their bodies, gradually preparing them for their conversion.
`11` Malmortis says: This special preparation grants them unique powers far greater than they would otherwise know.
`12` Malmortis says: Soon, the master will grant them the dark gift, making them fit to serve the Lich King for eternity!
`13` Malmortis says: Stay for as long as you like, **$n**. Glory in the fruits of your labor!
`14` Malmortis says: Your service has been invaluable in fulfilling the master's plan. May you forever grow in power....
**Malmortis walks into the portal and teleports away**

The first thing that stands out is the name variable (**$n**) contained in lines 2 and 13 isn't expressed by the script correctly.

The second problem is that the script is missing the last critical part of the dialogue. *Before* Malmortis walks into the portal he is supposed to say:
> `15` Malmortis says: Farewell.

At which point, the quest criteria is supposed to be fulfilled *and then* Malmortis goes on his way.

**Comment by Treeston** [comment]
`$n` means that the chat message isn't being passed a target, that's a pretty common bug.

**Comment by tkrokli** [comment]
Easy to notice when you look at `target_type` in that script. Whoever wrote that script forgot to target player for those texts to work properly. Should be posted as a separate issue along with its own solution.

**Comment by jerbookins** [comment]
@Killyana I already did and you closed it referencing this issue. #20882

**Comment by tkrokli** [comment]
That is because the missing quest credit is caused by the current waypoint issues. Only the `$n` (creature_text) part can be solved by updating the creature SAI script.

**Comment by Jildor** [comment]
What is the current status of the issue?

**Comment by tkrokli** [comment]
Mostly unchanged, none of the recent commits have solved the issue.
There are minor improvements, but only cosmetic (less bad visuals).
All of us will notice when a real solution gets added to the source.

**Comment by cBonky** [comment]
quest: Massacre At Light's Point
http://www.wowhead.com/quest=12701/massacre-at-lights-point
Miner should pull that mine car toward Light's Point.
Currently it doesn't happen.
https://www.youtube.com/watch?v=73pPKcDgh4k
is still broken

3.3.5 branch
TrinityCore rev. 46fcbb42d2ad
World DB: TDB 335.63
Operating system: Windows 7 x64

**Comment by tkrokli** [comment]
To all of you who follow this issue: Please test https://github.com/TrinityCore/TrinityCore/pull/21313
and see if anything else breaks by applying the change of removing line 77 (+78)
https://github.com/TrinityCore/TrinityCore/blob/c22764c18f5e6244c465ea49c5a2daa5d1e8e308/src/server/game/AI/ScriptedAI/ScriptedEscortAI.cpp#L75-L80

**Comment by ccrs** [comment]
dont do that

**Comment by Wilcalaf** [comment]
https://www.bountysource.com/issues/49200814-core-movement-waypoint-system-issues

**Comment by Infinito47** [comment]
Added another $15 to the bounty.

**Comment by Treeston** [comment]
"vehicle keeps following idle movement" should be fixed for all cases in d02f481, please report back if it isn't

**Comment by Treeston** [comment]
(still investigating waypoint AI issues)

**Comment by Treeston** [comment]
* e4e6e22 and fe1ba18 are systemic adjustments that should make EscortAI work consistently again
  * bonus feature: a fix for that ages-old movement after vehicle enter bug
* b243a79 (Hadronox), 92a8f65 (Territorial Trespass) and bf3973d (CoT4) fix unrelated quirks that were nevertheless mentioned above

This should fix all issues listed here. If no further issues crop up in the next few days, I will close this as resolved.

**Comment by Tauriella** [comment]
[Quest] The Escape (863) does not work

Description:
Escort Wizzlecrank's shredder out of the Venture Co. drill site.

Current behaviour:
Everything goes fine until the point after Wizzlecrank's Shredder says: "Looks like we're out of the woods, eh? Wonder what this does…", he seems to wonder around then despawns.

**Comment by Infinito47** [comment]
@Treeston 

Amazing results so far! I've tested CoS and Azjol Nerub and encountered the following:

Arthas still seems to get stuck in the tunnel for about 3 minutes. After this he continues, and then gets stuck attacked by The Citizans, but does not attack back.
 ^ Not sure if this is related to the waypoint issue though.

Hadronox also seems to start now, but after Hadronox goes up the tunnel, everything despawns again.

**Comment by Treeston** [comment]
If Hadronox jumps "through" the floor, make sure you have mmaps enabled. I can't reproduce that.

Evade issue should be fixed now.

**Comment by Treeston** [comment]
Fixed 863 too, @Tauriella (though it was unrelated to the actual issue)

**Comment by Foereaper** [comment]
I was able to recreate Hadronox flying through the floor issue, however it was prior to your changes so I don't know if it is related at all, or if it will even occur at this point. It seems if a player ends up having an entry in the threat list, but not enough to pull aggro from the creatures around Hadronox, and run back up to the "web" above, whenever Hadronox focuses on the player (player gains aggro) she will fly through the air and up towards the player, completely ignoring the path going up to the web. I'd imagine this is more of a path finding issue though and is unrelated to the original issue.

**Comment by Treeston** [comment]
Ah, right, I remember now. The room has some mmaps issues, and 100 instead of 70yd was probably enough to make it trigger sometimes.

Been a while since I did the nerub rewrite.

**Comment by Treeston** [comment]
6fb0bc1 and 2cd5992 should combine to make most of the old SAI scripts misusing `_INVOKER` work again - though they are still bad and wrong and need to be replaced eventually (see 482328b).

This should resolve all regressions as an intermediary measure, so I am closing this.

**Comment by Infinito47** [comment]
Arthas in CoS is still not working properly. He gets stuck everywhere and keeps running back to attack respawning NPC's. Is this not related to the waypoint issue?

**Comment by Aokromes** [comment]
Arthas in CoS never worked properly on clean TC
https://github.com/TrinityCore/TrinityCore/pull/17860

**Comment by Infinito47** [comment]
@Aokromes  Yeah.. I am aware of that, it had a few issues, but I remember completing the dungeon multiple times without arthas getting stuck before the re-write.

**Comment by pedroevil51** [comment]
Still a issue, how do i add on the bounty?

**Comment by illfated** [comment]
You need to open a new issue to get a new bountysource page. This issue is closed and paid out:
(https://www.bountysource.com/issues/49200814-core-movement-waypoint-system-issues-45)
